src/layers/microscopy-viewer-layer/microscopy-viewer-layer.js

```diff
@@ -25,58 +22,29 @@ export class MicroscopyViewerLayer extends CompositeLayer {
   }
 
   updateState() {
-    /* eslint-disable no-bitwise */
-    if (
-      !this.state.connections ||
-      this.props.useTiff ^ this.state.isTiff ||
-      this.props.useZarr ^ this.state.isZarr
-    ) {
-      /* eslint-disable no-bitwise */
-      if (this.props.useTiff) {
-        initTiff({ ...this.props }).then(
-          ({ connections, minZoom, imageWidth, imageHeight, tileSize }) => {
-            this.setState({
-              connections,
-              minZoom,
-              imageWidth,
-              imageHeight,
-              tileSize,
-              pool: new Pool(),
-              isZarr: false,
-              isTiff: true
-            });
-          }
-        );
-      } else {
-        initZarr({ ...this.props }).then(
-          ({ connections, minZoom, imageWidth, imageHeight, tileSize }) => {
-            this.setState({
-              connections,
-              minZoom,
-              imageWidth,
-              imageHeight,
-              tileSize,
-              isZarr: true,
-              isTiff: false
-            });
-          }
-        );
-      }
+    if (this.props.loader) {
+      const {
+        imageWidth,
+        imageHeight,
+        tileSize,
+        minZoom,
+        usePool
+      } = this.props.loader.vivMetadata;
+
+      this.setState({
+        minZoom,
+        imageWidth,
+        imageHeight,
+        tileSize,
+        pool: usePool ? new Pool() : null
```

Comment:
Why can't the pool be internal to the loader?  Your comment suggests that you think this should happen too, so if you were looking for my support, you have it.

Reply:
Further, do these need to be part of state?  I am just thinking at a high level, if we're passing the loader down, then why pass metadata that can be inferred from the loader?  It would help simplify the props a fair bit allowing for cleaner code, I think :)  This also gets back to the idea of subclassing - If we subclass, I think it would be clearer that these are just part of the classes.

Reply:
I need to explore this more. Something odd was happening with lifetimes of these workers when they weren't instantiated in state.

Reply:
Moved